### DISCUSSION TITLE
problems w/ fit=contain for small images

### DISCUSSION THREAD

**Discussion by rrrpdx** [fancyapps/ui#258] [Q&A] [answered]
Hi,
   I am trying to create a simple image viewer with FancyBox. I'd like the images to be displayed as large as possible, when displaying within the browser window, or being displayed fullscreen. i have tried setting  the .fancybox__content size to either {width:100%, height:100%} or {width:100vw, height:100vh}, while having fit=contain. I want to show the entire original image with the correct aspect ratio. This appears to work fine if the image size is greater than the window size. However, when the image is smaller than the window size, it is not scaled up to fill the window. 
  Attached is a small html file that I am using to test. It should display two landscape and two portrait images. The first of each pair is small resolution, and the latter is large resolution. Ideally, both images in a pair would be displayed the same. Is there a way to do this?

[test_fancy.txt](https://github.com/fancyapps/ui/files/8210580/test_fancy.txt)

**Comment by fancyapps** [comment, marked as answer]
Hi,

So, you wish that images would fill all available screen while maintaining image aspect ratio? You could do the other way - set

```js
Image : {
  fit: 'cover'
}
```

but then overwrite CSS:

```css
.has-image[data-image-fit=cover]  .fancybox__image {
    object-fit: contain !important;
}
```


https://jsfiddle.net/gjf7d6rw/

**Reply by rrrpdx** [reply]
Perfect, thank you so much.
